src/main/scala/analysis/solvers/FixPointSolver.scala

```diff
@@ -225,10 +225,12 @@ trait PushDownWorklistFixPointFunctions[N, T, L <: Lattice[T]] extends LinkedHas
    */
   def propagate(y: T, m: N): Unit = {
     val xm = x(m)
-    val t = lattice.sublattice.lub(xm, y)
-    if (t != xm) {
-      add(m)
-      x += m -> t
+    if (xm != y) {
+      val t = lattice.sublattice.lub(xm, y)
+      if (t != xm) {
+        add(m)
+        x += m -> t
+      }
```

Comment:
I think this will require close inspection of its impact, given the issue with the similar change to the process method.

Reply:
This seems to be fine in practice, and makes sense theoretically as long as the lub method isn't doing anything it shouldn't.